src/codec.rs

```diff
@@ -184,16 +194,26 @@ impl<'de, T> ::serde::Deserialize<'de> for Compact<T> where T: ::serde::Deserial
 	}
 }
 
-impl_from_compact! { u8, u16, u32, u64, u128 }
-
 /// Trait that tells you if a given type can be encoded/decoded in a compact way.
 pub trait HasCompact: Sized {
-	/// The compact type; this can be 
-	type Type: Encode + Decode + From<Self> + Into<Self> + Clone + PartialEq + Eq + MaybeDebugSerde;
+	/// The compact type; this can be
+	type Type: for<'a> EncodeAsRef<'a, Self> + Decode + From<Self> + Into<Self> + Clone +
```

Comment:
Isn't it redundant to require `Into<Self>` when `From<Self>` is also required? Standard library contains a trivial implementation of `Into<B>` for every `A` if there exists an `impl From<A> for B`.